### ISSUE TITLE
Building on Windows 10 fails

### ISSUE THREAD

**Issue by marksandford** [Islandora-Devops/isle-buildkit#21]
./gradlew.bat build failing on Windows 10 Pro, build 19041 with message:

```
> Task :imagemagick:build FAILED

FAILURE: Build failed with an exception.`

* What went wrong:
Execution failed for task ':imagemagick:build'.
> Process 'command 'docker'' finished with non-zero exit value 1
```

$ java -version
java version "1.8.0_251"

$ docker-compose version
docker-compose version 1.25.5, build 8a1c60f6
docker-py version: 4.1.0
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019

$ docker -v
Docker version 19.03.8, build afacb8b

At Danny's suggestion, I enabled experimental features, but that didn't seem to help.

[Console output with --stacktrace](https://github.com/Islandora-Devops/isle-buildkit/files/4732055/gradlew_build.txt)

**Comment by nigelgbanks** [comment]
@marksandford could you also add run the following command and attach the output to this ticket.

```bash
docker info
```

**Comment by marksandford** [comment]
@nigelgbanks 
here you go:

$ docker info
Client:
 Debug Mode: false

Server:
 Containers: 0
  Running: 0
  Paused: 0
  Stopped: 0
 Images: 25
 Server Version: 19.03.8
 Storage Driver: overlay2
  Backing Filesystem: <unknown>
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7ad184331fa3e55e52b890ea95e65ba581ae3429
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 4.19.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 2
 Total Memory: 1.944GiB
 Name: docker-desktop
 ID: H5KL:MFWP:EUQK:HCVT:JZIN:LVQW:4SH4:U6O5:J6RW:RWB6:OM37:3G6T
 Docker Root Dir: /var/lib/docker
 Debug Mode: true
  File Descriptors: 38
  Goroutines: 49
  System Time: 2020-06-05T15:15:36.70433418Z
  EventsListeners: 3
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: true
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine

**Comment by nigelgbanks** [comment]
I can reproduce this locally, might be due to networking issues and having partial files downloaded into the cache that are not valid and prevent further commands from being executed due to the gpg key or checksum not matching the downloaded file.

```bash
#8 [stage-0 2/4] RUN --mount=id=downloads,type=cache,target=/opt/downloads ...
#8 0.295 --2020-06-05 16:56:04--  https://archive.apache.org/dist/lucene/solr/7.1.0/solr-7.1.0.tgz
#8 0.305 Resolving archive.apache.org... 138.201.131.134, 2a01:4f8:172:2ec5::2
#8 0.306 Connecting to archive.apache.org|138.201.131.134|:443... connected.
#8 0.405 HTTP request sent, awaiting response... 304 Not Modified
#8 0.457 File '/opt/downloads/solr-7.1.0.tgz' not modified on server. Omitting download.
#8 0.457
#8 0.459 --2020-06-05 16:56:04--  https://archive.apache.org/dist/lucene/solr/7.1.0/solr-7.1.0.tgz.asc
#8 0.472 Resolving archive.apache.org... 138.201.131.134, 2a01:4f8:172:2ec5::2
#8 0.473 Connecting to archive.apache.org|138.201.131.134|:443... connected.
#8 0.573 HTTP request sent, awaiting response... 304 Not Modified
#8 0.622 File '/opt/downloads/solr-7.1.0.tgz.asc' not modified on server. Omitting download.
#8 0.622
#8 0.821 gpg: key DC92616F177050F6: public key "Shalin Shekhar Mangar (CODE SIGNING KEY) <shalin@apache.org>" imported
#8 0.828 gpg: Total number processed: 1
#8 0.828 gpg:               imported: 1
#8 1.060 gpg: Signature made Fri Oct 13 17:57:57 2017 UTC
#8 1.061 gpg:                using RSA key 38D2EA16DDF5FC722EBC433FDC92616F177050F6
#8 1.061 gpg:                issuer "shalin@apache.org"
#8 1.061 gpg: BAD signature from "Shalin Shekhar Mangar (CODE SIGNING KEY) <shalin@apache.org>" [unknown]
#8 ERROR: executor failed running [/bin/sh -c SOLR_VERSION="7.1.0" &&     install-apache-service.sh         --name solr         --version "${SOLR_VERSION}"         --key "38D2EA16DDF5FC722EBC433FDC92616F177050F6"         --mirror "https://archive.apache.org/dist/lucene/solr/${SOLR_VERSION}"         --file "solr-${SOLR_VERSION}.tgz"         docs example licenses server/solr/configsets]: runc did not terminate sucessfully
------
 > [stage-0 2/4] RUN --mount=id=downloads,type=cache,target=/opt/downloads     SOLR_VERSION="7.1.0" &&     install-apache-service.sh         --name solr         --version "${SOLR_VERSION}"         --key "38D2EA16DDF5FC722EBC433FDC92616F177050F6"         --mirror "https://archive.apache.org/dist/lucene/solr/${SOLR_VERSION}"         --file "solr-${SOLR_VERSION}.tgz"         docs example licenses server/solr/configsets:
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = failed to build LLB: executor failed running [/bin/sh -c SOLR_VERSION="7.1.0" &&     install-apache-service.sh         --name solr         --version "${SOLR_VERSION}"         --key "38D2EA16DDF5FC722EBC433FDC92616F177050F6"         --mirror "https://archive.apache.org/dist/lucene/solr/${SOLR_VERSION}"         --file "solr-${SOLR_VERSION}.tgz"         docs example licenses server/solr/configsets]: runc did not terminate sucessfully

```

**Comment by dannylamb** [comment]
The things I would look for all look ok there.  You're using overlay2 and have experimental enabled.   Others will probably know more than me though.

As another sanity check, can you run `docker buildx` successfully?  I couldn't ever finish a build until that command became available.

**Comment by dannylamb** [comment]
Ha, shoulda refreshed my browser.  Please ignore.

**Comment by nigelgbanks** [comment]
There is also another issue the file APKBUILD can be checked out on windows with windows line endings which breaks it when abuild is trying to execute it. I'll post fixes to both issues by tomorrow.

**Comment by nigelgbanks** [comment]
Note to self update code to get more verbose output when building:

```bash
docker build --progress=plain
```

```bash
abuild -v
```

**Comment by nigelgbanks** [comment]
@marksandford in the meanwhile if you open imagemagick\build\APKBUILD in nodepad++ or some other app that can change line endings and change it to unix line endings you should be able to build.

**Comment by marksandford** [comment]
Still failing at the same point.  I went through and changed all line endings to LF just to be safe, but no luck.

**Comment by nigelgbanks** [comment]
@marksandford did you try with the pull request linked to this ticket?

**Comment by marksandford** [comment]
Yes, I did.  Bash for Windows seems to want to change line endings on pull and commit, but that can be changed.

I tried deleting the two .gradle directories and /build as well.  Still getting stuck at the same point, on 2 different Windows 10 machines.

**Comment by nigelgbanks** [comment]
Ah seems that changing the branches after cloning won't modify the line endings on windows in the work tree, without rebuilding the index and updating the work tree.

I've tested both of the following on Windows and it works:

```
git clone --single-branch --branch build-issues https://github.com/nigelgbanks/isle-buildkit.git test
cd test
./gradlew build
```

Alternatively one can also modify the work tree after changing branches.

```
git checkout build-issues
git rm --cached -r . 
git reset --hard
./gradlew build
```

Once this is merged into master, any fresh clones of the repository on windows will not have to do these extra steps.

**Comment by nigelgbanks** [comment]
@marksandford  could you try using the commands above to confirm?

**Comment by marksandford** [comment]
Those commands worked on both machines! Thanks!

**Comment by dannylamb** [comment]
Resolved via https://github.com/Islandora-Devops/isle-buildkit/commit/7fd44340ac41e09918f44fabdef7bc3a25a710c2
